frameworks/moveos-stdlib/sources/bcs.move

```diff
@@ -41,6 +46,327 @@ module moveos_std::bcs{
         from_bytes<address>(v)
     }
 
+    /// Read `address` value from the bcs-serialized bytes.
+    public fun peel_address(v: vector<u8>): address {
+        assert!(vector::length(&v) >= 32, ErrorInvalidLength);
+        let i = 0;
+        let addr_bytes = vector::empty<u8>();
+        while (i < 32) {
+            let byte = vector::pop_back(&mut v);
+            vector::push_back(&mut addr_bytes, byte);
+            i = i + 1;
+        };
+        to_address(addr_bytes)
+    }
```

Comment:
We should define a `BCS` struct and use the struct as the argument.

Reply:
> We should define a `BCS` struct and use the struct as the argument.

Okay.